File: Modificador de 4 lenguaje/js/claseabstractas/Bicicleta.js
const Vehiculo = require("./Vehiculo");

class Bicicleta extends Vehiculo {
    constructor(marca) {
        super();
        this.marca = marca;
    }

    mover() {
        return `La bicicleta ${this.marca} avanza pedaleando.`;
    }
}

module.exports = Bicicleta;

File: Modificador de 4 lenguaje/C#/ClasesAbstractas/ClasesAbstractas/Bicicleta.cs
namespace EjemploConstructores.ClasesAbstractas
{
    class Bicicleta : Vehiculo
    {
        // Constructor sobrecargado con int
        public Bicicleta(int ruedas) { Nombre = $"Bicicleta con {ruedas} ruedas"; }

        // Constructor de copia
        public Bicicleta(Bicicleta otra) { Nombre = otra.Nombre; }

        public override void Mover() => System.Console.WriteLine($"{Nombre} se mueve pedaleando.");
    }
}
